src/Middleware/RequestDecompression/src/IRequestDecompressionProvider.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.RequestDecompression;
+
+/// <summary>
+/// Used to examine requests to see if decompression should be used.
+/// </summary>
+public interface IRequestDecompressionProvider
```

Comment:
Does this interface really need the 3 separate methods? Right now the Content-Encoding header is checked multiple times and look-up for `IDecompressionProvider` id done twice in `IsContentEncodingSupported` and `GetDecompressionProvider`.

I think the `IDecompressionProvider? GetDecompressionProvider(HttpContext context)` method should be enough.

Reply:
I've moved consolidated all the checks into `GetDecompressionProvider` and removed the other methods on the interface.